WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift

```diff
@@ -0,0 +1,332 @@
+                DDLogSwift.logError("Could not sync sites: \(error)")
+                self?.configureNoResultsView()
+                self?.refreshControl.endRefreshing()
+                self?.isSyncing = false
+        }
+    }
+
+
+    func handleRefresh(sender: AnyObject) {
+        syncSites()
+    }
+
+
+    func refreshFollowedPosts() {
+        let service = ReaderSiteService(managedObjectContext: managedObjectContext())
+        service.syncPostsForFollowedSites()
+    }
+
+
+    func unfollowSiteAtIndexPath(indexPath: NSIndexPath) {
+        guard let site = tableViewHandler.resultsController.objectAtIndexPath(indexPath) as? ReaderSiteTopic else {
+            return
+        }
+
+        let service = ReaderTopicService(managedObjectContext: managedObjectContext())
+        service.toggleFollowingForSite(site, success: { [weak self] in
+            self?.syncSites()
+            self?.refreshFollowedPosts()
+
+            }) { (error) in
+                DDLogSwift.logError("Could not unfollow site: \(error)")
+                let title = NSLocalizedString("Could not Unfollow Site", comment: "Title of a prompt.")
+                let description = error.localizedDescription
+                let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+                let alert = UIAlertController(title: title, message: description, preferredStyle: .Alert)
+                alert.addCancelActionWithTitle(buttonTitle)
+                alert.presentFromRootViewController()
+        }
+    }
+
+
+    func followSite(site: String) {
+        guard let url = urlFromString(site) else {
+            let title = NSLocalizedString("Please enter a valid URL", comment: "Title of a prompt.")
+            let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+            let alert = UIAlertController(title: title, message: "", preferredStyle: .Alert)
+            alert.addCancelActionWithTitle(buttonTitle)
+            alert.presentFromRootViewController()
+            return
+        }
+
+        let service = ReaderSiteService(managedObjectContext: managedObjectContext())
+        service.followSiteByURL(url, success: { [weak self] in
+            let success = NSLocalizedString("Followed", comment: "User followed a site.")
+            SVProgressHUD.showSuccessWithStatus(success)
+            self?.syncSites()
+
+        }) { (error) in
+            DDLogSwift.logError("Could not follow site: \(error)")
+            let title = NSLocalizedString("Could not Follow Site", comment: "Title of a prompt.")
+            let description = error.localizedDescription
+            let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+            let alert = UIAlertController(title: title, message: description, preferredStyle: .Alert)
+            alert.addCancelActionWithTitle(buttonTitle)
+            alert.presentFromRootViewController()
```

Comment:
There are three blocks of code in a row here that perform almost-identical steps to display an alert with an OK button. Perhaps extract a private method like `func presentErrorAlertWithTitle(title: String, description: String)`? Then each call site could just become:

```
let title = NSLocalizedString("Could not Follow Site", comment: "Title of a prompt.")
self.showErrorAlertWithTitle(title, description: error.localizedDescription)
```